debuglog-plugin/src/test/kotlin/com/bnorm/debug/log/IrPluginTest.kt

```diff
@@ -53,30 +65,146 @@ fun doSomething() {
+            }
+            catch (Throwable t) {
+                System.out.println((Object)("⇠ doSomething [" + (Object)Duration.toString-impl(markNow.elapsedNow-UwyO8pc()) + "] = " + t));
+                throw t;
+            }
+        }
+    """.trimIndent(),
+
+      outputVerifyFunc = { out ->
+        assert(out.size == 8)
+        assert(out[0] == "⇢ greet(greeting=Hello, name=World)")
+        assert(out[1].matches("⇠ greet \\[\\d+(\\.\\d+)?ms] = Hello, World!".toRegex()))
+        assert(out[2] == "Hello, World!")
+        assert(out[3] == "⇢ greet(greeting=Hello, name=Kotlin IR)")
+        assert(out[4].matches("⇠ greet \\[\\d+(\\.\\d+)?ms] = Hello, Kotlin IR!".toRegex()))
+        assert(out[5] == "Hello, Kotlin IR!")
+        assert(out[6] == "⇢ doSomething()")
+        assert(out[7].matches("⇠ doSomething \\[\\d+(\\.\\d+)?ms]".toRegex()))
+      }
+    )
   }
 
   @Test
   fun `IR plugin disabled`() {
-    val result = compile(sourceFile = main, DebugLogComponentRegistrar(false))
+    testIrPlugin(
+      isPluginEnabled = false,
+      expectedGreetMethod = """
+        public static final String greet(@NotNull final String greeting, @NotNull final String name) {
+            Intrinsics.checkNotNullParameter(greeting, "greeting");
+            Intrinsics.checkNotNullParameter(name, "name");
+            if (!Intrinsics.areEqual(greeting, "Hello")) {
+                return "Early return";
+            }
+            Thread.sleep(15L);
+            return greeting + ", " + name + '!';
+        }
+      """.trimIndent(),
+
+      expectedDoSomethingMethod = """
+        public static final void doSomething() {
+            Thread.sleep(15L);
+        }
+    """.trimIndent(),
+
+      outputVerifyFunc = { out ->
+        assertTrue(out.size == 2)
+        assertEquals("Hello, World!", out[0])
+        assertEquals("Hello, Kotlin IR!", out[1])
+      }
+    )
+  }
+
+  private fun testIrPlugin(
+    isPluginEnabled: Boolean,
+    expectedGreetMethod: String,
+    expectedDoSomethingMethod: String,
+    outputVerifyFunc: (List<String>) -> Unit
+  ) {
+    val result = compile(sourceFile = main, DebugLogComponentRegistrar(isPluginEnabled))
     assertEquals(KotlinCompilation.ExitCode.OK, result.exitCode)
 
-    val out = invokeMain(result, "MainKt").trim().split("""\r?\n+""".toRegex())
-    assertTrue(out.size == 2)
-    assertTrue(out[0] == "Hello, World!")
-    assertTrue(out[1] == "Hello, Kotlin IR!")
+    result.javaCode("MainKt").also { javaCode ->
+      assertMethod(expectedGreetMethod, javaCode, "public static final String greet")
+      assertMethod(expectedDoSomethingMethod, javaCode, "public static final void doSomething")
+    }
+
+    outputVerifyFunc(invokeMain(result, "MainKt").trim().split("""\r?\n+""".toRegex()))
+  }
+
+  private fun assertMethod(expectedMethod: String, javaCode: String, actualMethod: String) {
```

Comment:
I'd probably swap some parameters around a little bit and prefer function (more Kotlin) over method (more Java). It will probably read better as an ad-hoc assertion function this way.

```suggestion
  private fun assertFunction(javaCode: String, functionStatement: String, expectedFunction: String) {
```

Reply:
done